website/common/script/cron.js

```diff
@@ -88,37 +91,56 @@ export function daysSince (yesterday, options = {}) {
   Should the user do this task on this date, given the task's repeat options and user.preferences.dayStart?
  */
 
-export function shouldDo (day, dailyTask, options = {}) {
+export function shouldDo (day, dailyTask) {
   if (dailyTask.type !== 'daily') {
     return false;
   }
-  let o = sanitizeOptions(options);
-  let startOfDayWithCDSTime = startOfDay(_.defaults({ now: day }, o));
 
-  // The time portion of the Start Date is never visible to or modifiable by the user so we must ignore it.
-  // Therefore, we must also ignore the time portion of the user's day start (startOfDayWithCDSTime), otherwise the date comparison will be wrong for some times.
-  // NB: The user's day start date has already been converted to the PREVIOUS day's date if the time portion was before CDS.
-  let taskStartDate = moment(dailyTask.startDate).zone(o.timezoneOffset);
+  let daysOfTheWeek = [];
 
-  taskStartDate = moment(taskStartDate).startOf('day');
-  if (taskStartDate > startOfDayWithCDSTime.startOf('day')) {
-    return false; // Daily starts in the future
+  if (dailyTask.repeat) {
+    for (let [repeatDay, active] of Object.entries(dailyTask.repeat)) {
```

Comment:
is object.entries supported by our current babel presets? It's a stage 4 proposal that'll be included in es2017